packages/ra-core/src/store/localStorageStore.spec.ts

```diff
@@ -40,4 +40,14 @@ describe('localStorageStore', () => {
             expect(store.getItem('foo')).toEqual(undefined);
         });
     });
+    describe('reset preserving items outside Ra-Store', () => {
+        it('should reset the store', () => {
+            const store = localStorageStore();
+            store.setItem('foo', 'bar');
+            getStorage().setItem('baz', 'baz'); //set custom item in localstorage
+            store.reset();
+            expect(getStorage().getItem('baz')).toEqual('baz'); //expect not to be wiped on store reset
+            expect(store.getItem('foo')).toEqual(undefined);
+        });
+    });
```

Comment:
This test is good (and should be kept), but you are actually not testing the code you added. You should add a test about version change.